partner_firstname/tests/test_create.py

```diff
@@ -30,26 +30,25 @@ def tearDown(self):
         )
         for key, value in self.good_values.items():
             self.assertEqual(self.record[key], value, "Checking key %s" % key)
-
         super(PersonCase, self).tearDown()
 
     def test_no_name(self):
         """Name is calculated."""
         del self.values["name"]
 
     def test_wrong_name_value(self):
-        """Wrong name value is ignored, name is calculated."""
+        """Wrong name value is ignored, name is calculated for persons."""
         self.values["name"] = "BÄD"
 
     def test_wrong_name_context(self):
-        """Wrong name context is ignored, name is calculated."""
+        """Wrong name context is ignored, name is calculated for persons."""
         del self.values["name"]
         self.context["default_name"] = "BÄD"
 
     def test_wrong_name_value_and_context(self):
-        """Wrong name value and context is ignored, name is calculated."""
-        self.values["name"] = "BÄD1"
-        self.context["default_name"] = "BÄD2"
+        """Wrong name value and context is ignored, name is calculated for persons."""
```

Comment:
@pankk What' wrong with this this tests? Why do you modify the `setUp` method into the `CompanyCase` class? I don't see any specific case for the changes you introduce (at least it's not transparent into the code). The changes into `res_partner` make sense but not the changes into the test class. For example, two different values must be assigned to the name and the default_name into the context in `test_wrong_name_value_and_context`. This allow to detect what part of the code is broken if the the test fails.

Reply:
I tried to explain it in my earlier message. From memory: the unmodified tests will fail with the changes to the code, as the tests assume there will always be a firstname and a lastname component. However, when the partner type is a company only the name field is used (and lastname is computed using it, but firstname stays empty). It comes down to that for a company the name really does not need to be calculated, as it's already the only thing filled in, and firstname and lastname are not relevant in this case.

Reply:
@pankk `test_wrong_name_value_and_context`should not be modified. Only the definition of `good_values` into `CompanyCase` should be modified. Moreover, the expected `good_values' must be based on `self.values["name"]`. `self.good_values.update(lastname=self.values["name"], firstname=False, name=self.values["name"])`

Reply:
I didn't see an easy way to do it, hence my initial comment. My approach was to have something that worked without refactoring a large part of the module. Maybe it makes more sense to just replace the entire "company" case from the tests. The only thing in that case that should be tested is if the compute method does not touch the name.
I have no urgency to get this merged, yet it would be great if someone else could pick this up.

Reply:
I prefer separate test cases for company... IMO it's the only way to properly test cases related to company.

Reply:
Ellipsis not required. We're in agreement.